packages/typescript-estree/src/convert.ts

```diff
@@ -787,6 +788,14 @@ export class Converter {
       }
 
       case SyntaxKind.Identifier: {
+        if (isThisInTypeQuery(node)) {
+          return this.createNode<TSESTree.ThisExpression>(
+            node as unknown as ts.ThisExpression,
```

Comment:
this casting is incorrect, we should fix that

Reply:
you have to add correct type to `EstreeToTsNodeTypes`, and remove this casting

https://github.com/typescript-eslint/typescript-eslint/blob/74c67ba1434291f429c02db6ce72418c512dc07f/packages/typescript-estree/src/ts-estree/estree-to-ts-node-types.ts#L71-L74